dapp/src/components/dialogs/DeleteNameDialog.tsx

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+'use client';
+
+import { Info } from '@iota/apps-ui-icons';
+import {
+    Button,
+    Dialog,
+    DialogBody,
+    DialogContent,
+    Header,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    LoadingIndicator,
+} from '@iota/apps-ui-kit';
+import { useCurrentAccount, useIotaClient, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { isSubName } from '@iota/iota-names-sdk';
+import { useMutation } from '@tanstack/react-query';
+
+import { NameUpdate, useRegistrationNfts, useUpdateNameTransaction } from '@/hooks';
+import { RegistrationNft } from '@/lib/interfaces/registration.interfaces';
+import { getSubdomainObjectId } from '@/lib/utils/names';
+
+type DeleteNameDialogProps = {
+    nft: RegistrationNft;
+    open: boolean;
+    setOpen: (bool: boolean) => void;
+};
+
+export function DeleteNameDialog({ nft, open, setOpen }: DeleteNameDialogProps) {
+    const account = useCurrentAccount();
+    const iotaClient = useIotaClient();
+    const { data: subdomainsOwned } = useRegistrationNfts('subdomain');
+
+    const isNameSubName = nft ? isSubName(nft.name) : null;
+    const isExpired = nft.isExpired || false;
+
+    // Create updates
+    const updates: NameUpdate[] = [];
+
+    if (nft && isExpired) {
+        const nftid = isNameSubName
+            ? (getSubdomainObjectId(subdomainsOwned ?? [], nft.name) ?? '')
+            : nft.id;
+
+        updates.push({
+            type: 'delete-name',
+            nft: nftid,
+            isSubname: isNameSubName || false,
+        });
```

Comment:
Let's not pass empty ids
```suggestion
        const nftId = isNameSubName
            ? getSubdomainObjectId(subdomainsOwned ?? [], nft.name)
            : nft.id;
            
        if(nftId) {
        	updates.push({
	            type: 'delete-name',
	            nft: nftid,
	            isSubname: isNameSubName || false,
	        });
        }
```